### PR TITLE
Allow JsonObject and JsonArray to be used in any POJO for JSON handling

### LINKED ISSUE #39599: JsonObject is empty when used with resteasy-reactive
### Describe the bug

I have the following code:

```java
    import io.vertx.core.json.JsonObject;

    public class CreateSomethingRequest {
      @NotNull
      public JsonObject payload;

      @Size(max = 10)
       public List<@Valid Attachment> attachements;
   }

@Path ("/path")
public class Controller {
    @POST
    public RestResponse<Map<String, String>> createSomething(@Valid CreateSomethingRequest request) {
        var id = ....
        return RestResponse.status(CREATED, Map.of("somethingID", id));
    }
}
```

When I make the following POST request:

image

And when I debug it in IntelliJ, then I see that the payload is empty:
image

If I replace `JsonObject` with `Map<String, Object>`, then it starts working (again): 

image

Strangely, the validation is also not happening (or skipped), since the `payload` field is annotated with `@NotNull`.

Note that this used to work in Quarkus 3.2.x. Since recently we migrated to 3.8.2 and we see this odd behaviour happening.

And another odd thing is, that our integration test succeeds with RestAssured & Wiremock. This only happens if we startup the application (in dev or prd mode).

### Expected behavior

`JsonObject` should be mapped correctly.

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

Windows 10

### Output of `java -version`

jdsk 17 Temurin

### Quarkus version or git rev

3.8.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Maven 3.8.6

### Additional information

My pom.xml

```xml
<dependencies>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-smallrye-jwt</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-hibernate-validator</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-smallrye-openapi</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-rest-client-reactive-jackson</artifactId>
        </dependency>
        <dependency>
            <groupId>io.quarkus</groupId>
            <artifactId>quarkus-smallrye-health</artifactId>
        </dependency>
</dependencies>
```

### PR DESCRIPTION
**Pull request by geoand**
- Closes: #39599

### CODE REVIEW

**Review by FroMage — APPROVED** [review]

File: extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/jackson/JsonArrayDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.fasterxml.jackson.databind.deser.std.StdDelegatingDeserializer;
+import com.fasterxml.jackson.databind.util.Converter;
+import com.fasterxml.jackson.databind.util.StdConverter;
+
+import io.vertx.core.json.JsonArray;
+
+public class JsonArrayDeserializer extends StdDelegatingDeserializer<JsonArray> {
+
+    public JsonArrayDeserializer() {
+        super(new StdConverter<List<?>, JsonArray>() {
```
[the remark this answers is not in the document]
Comment (geoand) [reply extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/jackson/JsonArrayDeserializer.java:16]: Actually, what Jackson normally proposes is to not have a subclass, but to just pass the converter as a constructor argument.
We cannot do that however (at least not with some work which is of extremely limited usefulness) due to how the Jackson Module is created - we need a no-args constructor

**Comment by geoand** [comment]
I am not sure what the deal with `ImageMetricsITCase` is...

**Comment by geoand** [comment]
Thanks a lot for the analysis!

I'll update the PR tomorrow

**Comment by gsmet** [comment]
I think this threshold management is problematic meaning we are completly missing the PRs that made it go from `6048` to `6228` and we raise a problem for a PR that is (rightfully) adding 2 more classes.

I think we will have to find a better solution to actually catch the culprits and not only the PR exceeding the threshold for just a couple of classes.

**Comment by jerboaa** [comment]
> I think this threshold management is problematic meaning we are completly missing the PRs that made it go from `6048` to `6228` and we raise a problem for a PR that is (rightfully) adding 2 more classes.
> 
> I think we will have to find a better solution to actually catch the culprits and not only the PR exceeding the threshold for just a couple of classes.

Agreed. But it's better than not getting a signal at all. AFAIK, there was an [issue raised when the test would fail too often](https://github.com/quarkusio/quarkus/pull/39044). I'll do some digging and see if I can find out when the more significant bump happened and if I do, create an issue for it.

**Comment by gsmet** [comment]
One thing that we could do could be to extract these metrics to Develocity and add them as custom values to the build.

Apparently, they are planning some work to make it possible to extract this data easily and we could potentially have a graph with a link to the builds actually introducing the regression.

We would have to store this thing to a file though to be able to pass it to the build scan.

**Comment by jerboaa** [comment]
> We would have to store this thing to a file though to be able to pass it to the build scan.

The native builds produce `<image-name>-build-output-stats.json`, which are json files which have the data. The MetricsIT tests use those files and the configured thresholds to fail the test. Is that what you'd need in terms of files?

**Comment by gsmet** [comment]
Yeah, I suppose we could write something that extracts the data from these files and push it to the build scans.

**Review by cescoffier — APPROVED** [review]

**Comment by jerboaa** [comment]
> > I think this threshold management is problematic meaning we are completly missing the PRs that made it go from `6048` to `6228` and we raise a problem for a PR that is (rightfully) adding 2 more classes.

Correction. I've rerun the test with `Mandrel-23.1.1.0-Final`, `Mandrel-23.1.2.0-Final`, `Mandrel-23.1.1.0-Final` and they all have `6156` reachable reflective types. So not sure where the initial threshold of `6048` comes from.

Either way, we have increased the image size from 612d8169ebfeb1e082b499c910474934554e9a60 to e7f2a1ee4aff4c5efeba2d414fb696d0ac47f1f4 from `80.31MB` to `81.05MB`. Death by a thousand cuts. I'll file an issue to track this.

**Comment by jerboaa** [comment]
> I'll file an issue to track this.

Tracking issue is: https://github.com/quarkusio/quarkus/issues/39674

**Pull request merged**